src/tools/rbd_mirror/PoolReplayer.h

```diff
@@ -60,6 +67,8 @@ class PoolReplayer {
   bool is_running() const;
 
   void init(const std::string& site_name);
+  int update_peers(const Peers& peers);
+  int add_peer(const PeerSpec& peer_spec, C_Gather* gather_ctx);
```

Comment:
Just wondering, couldn't we make `add/remove_peer` "non blocking". I.e. it would just update the peers list down to the image replayer and set a flag for the image replayer that it needs restarting to apply the change?

Reply:
I use the add_peer to init some state like the StatusUpdater and the PoolWatcher that's why it's blocking right now. I could make it non blocking but this would require some sort of reconciliation logic to "init" some ressource for the peer and propagate the add_peer to the other components once this is done.